### PR TITLE
Generate docs and source JAR for jazzer-junit

### LINKED ISSUE #474: JUnit source JAR on Maven Central is invalid
The new JUnit integration is awesome, but it looks like the source JAR uploaded to Maven Central (jazzer-junit-0.12.0-sources.jar) is invalid. My IDE (Eclipse) is unable to use it to browse the sources, and 7Zip is unable to open it (which is usually possible).

### PR DESCRIPTION
**Pull request by fmeum**
The automatic docs and source JAR generation provided by `rules_jvm_external`'s `java_export` couldn't handle our shading and stripping.

Fixes #474

### CODE REVIEW

**Comment by fmeum** [comment]
@bertschneider I pushed a new commit that I hope fixes the flakiness for good. I observed that the test always passed on retry, so running it by itself should work well.

File: driver/src/main/java/com/code_intelligence/jazzer/junit/BUILD.bazel [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
+# The Jazzer JUnit integration without its dependencies declared in the pom.xml, together with the unshaded sources.
+java_import(
     name = "junit",
+    jars = ["junit_shaded.jar"],
+    srcjar = "junit_unshaded_deploy-src.jar",
+    visibility = ["//visibility:public"],
+)
+
+javadoc(
+    name = "junit-docs",
```
Comment (bertschneider) [inline driver/src/main/java/com/code_intelligence/jazzer/junit/BUILD.bazel:18]: The `@FuzzTest` Javadoc seems outdated.

**Review by bertschneider — COMMENTED** [review]
Changing `JAZZER_VALUE_PROFILE` to `True` and `False`, as suggested by the error message, fixes the tests for me.

File: driver/src/main/java/com/code_intelligence/jazzer/junit/BUILD.bazel (hunk 1 above)
Comment (fmeum) [reply driver/src/main/java/com/code_intelligence/jazzer/junit/BUILD.bazel:18]: Yes, I want to make one more breaking change to the API before I update it.

**Comment by fmeum** [comment]
@bertschneider Do you know whether Javadocs for one Maven dependency can reference classes from another one? Otherwise we should probably include the docs for `jazzer-api` in those for `jazzer-junit`.

**Comment by bertschneider** [comment]
Not that I'm aware of.

**Review by bertschneider — APPROVED** [review]

**Comment by fmeum** [comment]
> Not that I'm aware of.

I have inlined the docs for the non-hook part of the API into the `jazzer-junit` docs.

**Pull request merged**